dashboard/src/modules/components/TableComponent/index.jsx

```diff
@@ -0,0 +1,233 @@
+import SearchBox from "../SearchComponent";
+import DatePickerWidget from "../DatePickerComponent";
+import Heading from "../HeadingComponent";
+import PathBreadCrumb from "../BreadCrumbComponent";
+import AlertMessage from "../AlertComponent";
+import NavItems from "../NavbarComponent";
+import EmptyTable from "../EmptyStateComponent";
+import moment from "moment";
+import { fetchPublicDatasets } from "../../../actions/fetchPublicDatasets";
+import TablePagination from "../PaginationComponent";
+let startDate = moment(new Date(1990, 10, 4)).format("YYYY-MM-DD");
+let endDate = moment(new Date()).format("YYYY-MM-DD");
+let controllerName = "";
+let dataArray = [];
+export const TableWithFavorite = () => {
+  const columnNames = {
+    controller: "Controller",
+    name: "Name",
+    creationDate: "Created On",
+  };
+  const [activeSortIndex, setActiveSortIndex] = useState(null);
+  const [activeSortDirection, setActiveSortDirection] = useState(null);
+  const [favoriteRepoNames, setFavoriteRepoNames] = useState([]);
+  const [publicData, setPublicData] = useState([]);
+  const [isSelected, setIsSelected] = useState("controllerListButton");
+  const [isNavOpen, setIsNavOpen] = useState(false);
+  const [page,setPage]=useState(1);
+  const [perPage,setPerPage]=useState(10);
+  const dispatch=useDispatch();
+  useEffect(() => {
+     dispatch(fetchPublicDatasets()).then((res) => {
+        dataArray = res.data;
+        setPublicData(res.data);
+        setFavoriteRepoNames(JSON.parse(localStorage.getItem("favControllers")))
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }, []);
+  const markRepoFavorited = (repo, isFavoriting = true) =>{
+   const otherFavorites = favoriteRepoNames.filter((r) => r.name !== repo.name);
+      const newFavorite= isFavoriting ? [...otherFavorites, repo] : otherFavorites;
+      saveFavorites(newFavorite);
+      setFavoriteRepoNames(newFavorite);
+  }
+    const selectedArray =isSelected === "controllerListButton" ? publicData.slice((page-1)*(perPage),(page*perPage)) : favoriteRepoNames.slice((page-1)*(perPage),(page*perPage));
+  const onNavToggle = () => {
+    setIsNavOpen(!isNavOpen);
+  };
+  const isRepoFavorited = (repo) => {
+    for (let i = 0; i < favoriteRepoNames.length; i++) {
+      if (
+        repo.name === favoriteRepoNames[i].name &&
+        repo.controller === favoriteRepoNames[i].controller &&
+        repo.metadata["dataset.created"] ===
+          favoriteRepoNames[i].metadata["dataset.created"]
+      )
+        return true;
+    }
+    return false;
+  };
+  const getSortableRowValues = (publicData) => {
+    const { controller, name } = publicData;
+    const creationDate = publicData.metadata["dataset.created"];
+    return [controller, name, creationDate];
+  };
+  let sortedRepositories = selectedArray;
+  if (activeSortIndex !== null) {
+    sortedRepositories = selectedArray.sort((a, b) => {
+      const aValue = getSortableRowValues(a)[activeSortIndex];
```

Comment:
LGTM thinks that [`sortedRepositories` isn't actually used anywhere](https://lgtm.com/projects/g/distributed-system-analysis/pbench/rev/pr-9039697a31ff406383a8b760c96ce53455604f7a); and, if it's wrong, I don't see that either. Can this code be cleaned up?